### ISSUE TITLE
Deserialization crashes app with assert on static array

### ISSUE THREAD

**Issue by TETYYS** [vibe-d/vibe.d#1942]
If, for example we register an REST interface with a static array as a parameter, a remote request can crash the application by passing an incorrect length array.

Crash occurs here: https://github.com/vibe-d/vibe.d/blob/master/data/vibe/data/serialization.d#L645 on `sz == T.length`

**Comment by bausshf** [comment]
I don't think this is really an issue, because it's an **assert** statement and thus it only happens during debugging. During release builds it won't crash, because the assert is omitted.

The assert makes sense, because it helps you catch internal bugs where you pass arrays with invalid lengths.

Asserts are not meant to be used for input validation. You use **enforce** for that.

**Comment by Geod24** [comment]
The code in question was https://github.com/vibe-d/vibe.d/blob/714e6bc63bf2ff4c8ee9794934eecbc64f0d9a04/data/vibe/data/serialization.d#L645-L651

Nowadays it's: https://github.com/vibe-d/vibe.d/blob/d638b343d6f9ac8da5b18dff61c772d92aa31aa3/data/vibe/data/serialization.d#L766-L778

The `assert` is still a problem, because not all servers use `-release`.
